topi/python/topi/arm_cpu/conv2d.py

```diff
@@ -243,6 +251,94 @@ def _decl_spatial_pack(cfg, data, kernel, strides, padding, dilation, layout, ou
                          name='output_unpack', tag='spatial_conv2d_output')
     return output
 
+
+def _decl_spatial_pack_nhwc(cfg, data, kernel, strides, padding, dilation,
+                            layout, out_dtype, num_tile):
+    assert layout == "NHWC", "Only support NHWC"
+    # create workload according to raw arguments
+    out_dtype = out_dtype or data.dtype
+    N, IH, IW, CI = get_const_tuple(data.shape)
+
+    # TODO? Dilation
+
+    if len(kernel.shape) == 4:
+        pre_packed = False
+        KH, KW, _, CO = get_const_tuple(kernel.shape)
+    else:
+        pre_packed = True
+        CO, _, KH, KW, VC = get_const_tuple(kernel.shape)
+        CO = CO * VC
+
+    pad_top, pad_left, pad_bottom, pad_right = get_pad_tuple(padding, (KH, KW))
+    HSTR, WSTR = strides if isinstance(strides, (tuple, list)) else (strides, strides)
+    OH = (IH + pad_top + pad_bottom - KH) // HSTR + 1
+    OW = (IW + pad_left + pad_right - KW) // WSTR + 1
+    data_pad = pad(data, [0, pad_top, pad_left, 0], [0, pad_bottom, pad_right, 0])
+
+    # ==================== define configuration space ====================
+    n, co, oh, ow = cfg.axis(N), cfg.axis(CO), cfg.axis(OH), cfg.axis(OW)
+    ci, kh, kw = cfg.reduce_axis(CI), cfg.reduce_axis(KH), cfg.reduce_axis(KW)
+
+    if num_tile == 2:     # for arm cpu
+        co, vc = cfg.define_split('tile_co', co, num_outputs=2)
+        oh, vh = cfg.define_split('tile_oh', oh, num_outputs=2)
+        ow, vw = cfg.define_split('tile_ow', ow, num_outputs=2)
+    elif num_tile == 3:   # for mali gpu
+        co, _, vc = cfg.define_split('tile_co', co, num_outputs=3)
+        oh, _, vh = cfg.define_split('tile_oh', oh, num_outputs=3)
+        ow, _, vw = cfg.define_split('tile_ow', ow, num_outputs=3)
+    else:
+        raise RuntimeError("Invalid num_tile")
+
+    cfg.define_reorder("reorder_0",
+                       [n, oh, ow, co, ci, kh, kw, vh, vc, vw],
+                       policy='candidate', candidate=[
+                           [n, oh, ow, co, ci, kh, kw, vh, vc, vw],
+                           [n, oh, ow, co, ci, kh, kw, vc, vh, vw]])
+
+    cfg.define_annotate("ann_reduce", [kh, kw], policy='try_unroll')
+    cfg.define_annotate("ann_spatial", [vh, vw, vc], policy='try_unroll_vec')
+
+    # fallback support
```

Comment:
Cleanup

Reply:
Good catch, this shouldnt have been left commented out. It's now fixed.